doc/reference/configuration/configuration_reference.rst

```diff
@@ -3860,6 +4145,78 @@ The ``replication`` section defines configuration parameters related to :ref:`re
     | Default: 0
     | Environment variable: TT_REPLICATION_SYNC_TIMEOUT
 
+
+.. _configuration_reference_replication_synchro_queue_max_size:
+
+.. confval:: replication.synchro_queue_max_size
+
+    **Since:** :doc:`3.3.0 </release/3.3.0>`
+
+    The ``replication.synchro_queue_max_size`` option controls the maximum size of the synchronous transaction queue on a master node.
+    This parameter ensures that the queue does not grow indefinitely, potentially impacting performance and resource usage.
+
+    The size is measured in bytes, with a value of 0 disabling the limit (unlimited queue, as in earlier versions).
+
+    .. NOTE::
+
+        Applies only to the master node.
+
+
+    If the synchronous queue reaches the configured size limit, new transactions attempting to enter the queue are discarded.
+    In such cases, the system returns an error to the user:
+    “The synchronous transaction queue is full.”
+
+    This size limitation does not apply during the recovery process. Transactions processed during recovery are unaffected by the queue size limit.
+
+    Use the following command to view the current size of the synchronous queue:
+
+    .. code-block:: lua
+
+        box.info.synchro.queue.size
+
+
+    .. code-block:: command line
+
+        tarantool> box.info.synchro
+        ---
+        - queue:
+            owner: 1
+            size: 60
+            busy: false
+            len: 1
+            term: 2
+            quorum: 2
+        ...
+
+    Set the synchronous queue size limit in the configuration file:
+
+    .. code-block:: yaml
+
+        replication:
+            synchro_queue_max_size: 33554432  # Limit set to 32 MB
+
+
+    .. code-block:: lua
+
+        box.cfg{
+          replication_synchro_queue_max_size = 33554432  -- Limit set to 32 MB
+        }
+
+
+    .. hint::
```

Comment:
No such admonition type in Tarantool docs. If it's useful in general, discuss and approve the usage guidelines with the whole team.

Also, the thoughts in this block don't seem new/unique. Everything is already said above, so why repeat them?

Reply:
thanks, fixed